### PR TITLE
Avocado.skip* usable on classes

### PR DESCRIPTION
**Pull request by richtja**
This PR expands the avocado.skip* decorators from methods to classes.
When a user uses skip decorator on the class, the decorator is applied
to every class method.

Reference: https://github.com/avocado-framework/avocado/issues/3473
Signed-off-by: Jan Richter <jarichte@redhat.com>

### CODE REVIEW

File: avocado/core/decorators.py [hunk 1]
```diff
@@ -73,14 +73,28 @@ def skip(message=None):
...
+                    raise core_exceptions.TestSkipError(message)
+                function = wrapper
+            function.__skip_test_decorator__ = True
+            return function
+
+        def class_decorator(cls):
+            for key in cls.__dict__:
+                value = getattr(cls, key)
+                if callable(getattr(cls, key)):
+                    wrapped = method_decorator(value)
```
Comment (willianrampazzo) [inline avocado/core/decorators.py]: I don't think I understood the need to set `value` before the `if` statement. Can't you do: 

```
if callable(getattr(cls, key)):
    wrapped = method_decorator(getattr(cls, key))
```
or, using the `value` variable:

```
value = getattr(cls, key)
if callable(value):
    wrapped = method_decorator(value)
```
Maybe I'm missing something here :)

File: avocado/core/decorators.py [hunk 2]
```diff
@@ -73,14 +73,28 @@ def skip(message=None):
...
+                    setattr(cls, key, wrapped)
+            return cls
+
+        if inspect.isclass(obj):
```
Comment (willianrampazzo) [inline avocado/core/decorators.py]: Looking at _inspect_ source (https://github.com/python/cpython/blob/master/Lib/inspect.py#L72), the method _isclass_ does `isinstance(object, type)`. In this case, I don't see a reason to use _inspect_, in favor of `isinstance(obj, type)`.

File: avocado/core/decorators.py (hunk 1 above)
Comment (richtja) [reply avocado/core/decorators.py]: Yes, you are right there is no need for setting `value`.

File: avocado/core/decorators.py (hunk 2 above)
Comment (richtja) [reply avocado/core/decorators.py]: Ok, I will change that.

**Comment by richtja** [comment]
@willianrampazzo thanks for the review, I will create a new version with fixes.

**Comment by richtja** [comment]
I'm closing because of #3570

**Pull request closed without merging**